src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Base64Validator.cs

```diff
@@ -0,0 +1,192 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace System.Buffers.Text
+{
+    public static partial class Base64
+    {
+        /// <summary>
+        /// Validates the span of UTF-8 encoded text represented as base64 into binary data.
+        /// </summary>
+        /// <param name="base64Text">The input span which contains UTF-8 encoded text in base64 that needs to be validated.</param>
+        /// <returns>true if <paramref name="base64Text"/> is decodable; otherwise, false.</returns>
+        public static bool IsValid(ReadOnlySpan<char> base64Text) => IsValid(base64Text, out int _);
+
+        /// <summary>
+        /// Validates the span of UTF-8 encoded text represented as base64 into binary data.
+        /// </summary>
+        /// <param name="base64Text">The input span which contains UTF-8 encoded text in base64 that needs to be validated.</param>
+        /// <param name="decodedLength">The maximum length (in bytes) if you were to decode the base 64 encoded text <paramref name="base64Text"/> within a byte span.</param>
+        /// <returns>true if <paramref name="base64Text"/> is decodable; otherwise, false.</returns>
+        public static unsafe bool IsValid(ReadOnlySpan<char> base64Text, out int decodedLength)
+        {
+            if (base64Text.IsEmpty)
+            {
+                decodedLength = 0;
+                return true;
+            }
+
+            // Check for invalid chars
+            int indexOfFirstNonBase64 = base64Text.IndexOfAnyExcept(ValidBase64CharsSortedAsc);
+            if (indexOfFirstNonBase64 > -1)
+            {
+                decodedLength = 0;
+                return false;
+            }
+
+            int length = base64Text.Length;
+            int paddingCount = 0;
+
+            // Check if there are chars that need to be ignored while determining the length
+            if (base64Text.IndexOfAny(CharsToIgnore) > -1)
```

Comment:
```suggestion
            if (base64Text.IndexOfAny(CharsToIgnore) >= 0)
```
And use `IndexOfAnyValues` (see above).